server/controllers/gift.controller.js

```diff
@@ -7,20 +7,19 @@ const Guest = DB.Guest;
 const { messages } = require('./../helper');
 
 const isEventOwner = (eventId, userId, gift) => {
-  return Event.findOne({where: {id: +eventId, $and: {owner: userId}}})
-  .then(event => !!event && !!gift && (+gift.dataValues.event_id === +eventId))
+  return Event.findOne({where: {id: eventId, $and: {owner: userId}}})
+  .then(event => !!event && !!gift)
   .catch(error => error);
 };
 
 const isEventGuest = (eventId, userId, gift, res) => {
-  return Guest.findOne({where: {event_id: +eventId, $and: {user_id: userId}}})
-  .then(guest => !!guest && !!gift && (+gift.dataValues.event_id === +eventId) ||
-    res.status(404).send(messages.giftNotFound))
+  return Guest.findOne({where: {event_id: eventId, $and: {user_id: userId}}})
```

Comment:
$and is a Sequelize query operator

http://docs.sequelizejs.com/manual/tutorial/querying.html